### DISCUSSION TITLE
Add an option to automatically switch between roll and auto modes

### DISCUSSION THREAD

**Discussion by mpekurin** [OpenHantek/OpenHantek6022#258] [Ideas]
It would be nice to have a checkbox (enable / disable) and an input field with a timebase threshold at which the trigger mode switches between roll (>= threshold) and auto (< threshold). I guess the best place for these things will be the Trigger section.

From my personal experience I may suggest the default value of the threshold to be 20 ms.

**Comment by Ho-Ro** [comment]
The scroll mode is more of a quick'n'dirty hack than a clean implementation, but it works as it is. Because of the less than perfect implementation, automatic switching would require a lot of cross-module communication (signal-slot etc.) - on the other hand, I don't want to invest too much time in a clean refactoring, but can of course give hints. So unfortunately you are on your own if you want to implement this or wait for a volunteer.
